csrc/python_frontend/python_bindings.cpp

```diff
@@ -3148,6 +3148,41 @@ void initNvFuserPythonBindings(PyObject* module) {
       },
       py::arg("arg"),
       py::arg("dim"));
+  nvf_sched.def(
+      "reduction_factor",
+      [](FusionDefinition::SchedOperators& self,
+         Tensor arg,
+         std::vector<int>& dims) -> Tensor {
+        FUSER_PERF_SCOPE("SchedOperators.reduction_factor");
+        TORCH_CHECK(
+            self.validUse(),
+            "Attempting to use a SchedOperators Op prior to definition!");
+        FusionDefinition* fd = self.fusion_definition;
+        auto input_tv =
+            fd->getFusionState(arg.index)->template as<TensorView>();
+        auto output_tv = input_tv->rFactor(dims);
+        Tensor output = fd->defineTensor(arg.dims);
+        fd->addFusionState(output.index, output_tv);
```

Comment:
I suggest that we do the check here instead.
check the fusion state length before and after adding output_tv to the fusion state.